tests/linear_model/test_glm.py

```diff
@@ -203,3 +220,79 @@ def test_logistic_predict_proba_shape():
     lr.fit(X, y)
     prob = lr.predict_proba(X)
     assert prob.shape == (100, 2)
+
+@pytest.mark.parametrize(
+    "reg_type,data_generator", 
+    [(LinearRegression, "single_chunk_regression"), 
+    (LogisticRegression, "single_chunk_classification"),
+    (PoissonRegression, "single_chunk_count_classification")
+    ]
+)
+def test_model_coef_dask_numpy(reg_type, data_generator, request):
+    """Tests that models return same coefficients and intercepts with array types"""
+    X, y = request.getfixturevalue(data_generator)
+    np_mod, da_mod = reg_type(fit_intercept=True), reg_type(fit_intercept=True)
+    da_mod.fit(X, y)
+    np_mod.fit(X.compute(), y.compute())
+    np.testing.assert_allclose(da_mod.coef_, np_mod.coef_)
+    np.testing.assert_allclose(da_mod.intercept_, np_mod.intercept_)
+
+@pytest.mark.parametrize(
+    "reg_type, skl_reg_type, skl_params, data_generator", 
+    [(LinearRegression, sklLinearRegression, {}, "single_chunk_regression"), 
+    (LogisticRegression, sklLogisticRegression, {'penalty' : 'none'}, "single_chunk_classification"),
+    (PoissonRegression, sklPoissonRegressor, {'alpha' : 0}, "single_chunk_count_classification")
+    ]
+)
+def test_model_against_sklearn(reg_type, skl_reg_type, skl_params, data_generator, request):
+    """
+    Tests accuracy of model predictions
+
+    Notes:
+        1. Dask ML 'newton' solver does not implement a penalty, only
+            set arbitarily high for clarity of non-regularization.
+        2. This method only works for `fit_intercept=False`
+    """
+    X, y = request.getfixturevalue(data_generator) 
+    dask_ml_model = reg_type(fit_intercept=True, solver='newton', penalty='l2', C=1e8)
+    skl_model = skl_reg_type(fit_intercept=True, **skl_params)
+    # skl_model has to be fit with numpy data
+    dask_ml_model.fit(X, y)
+    skl_model.fit(X.compute(), y.compute())
+    np.testing.assert_allclose(dask_ml_model.predict(X), skl_model.predict(X), rtol=1e-3, atol=1e-3)
+
+
+@pytest.mark.parametrize(
+    "reg_type, skl_reg_type, skl_params, dataset_function", 
+    [(LinearRegression, sklLinearRegression, {}, make_regression), 
+    (LogisticRegression, sklLogisticRegression, {'penalty' : 'none'}, make_classification),
+    ]
+)
+def test_model_coef_against_sklearn(reg_type, skl_reg_type, skl_params, dataset_function):
+    """
+    Tests accuracy of coefficient calculation.
+
+    sk-learn least squares method does not tie out to iterative sovlers with
+    e.g. Rdi
+    """
+    X, y = dataset_function(n_samples=100, n_features=20, n_informative=10, chunks=100, random_state=0)
+    X_np, y_np = X.compute(), y.compute()
+    
+    skl_model = skl_reg_type(fit_intercept=True, **skl_params)
+    skl_model.fit(X_np, y_np)
+
+    dask_ml_model = reg_type(fit_intercept=True, solver='newton', penalty='l2', C=1e8)
+    dask_ml_model.fit(X, y)
+
+    np.testing.assert_allclose(skl_model.coef_.flatten(), dask_ml_model.coef_, rtol=1e-3, atol=1e-3)
```

Comment:
Maybe add a test to make sure that both elements are approximately equal?

```suggestion
    np.testing.assert_allclose(skl_model.coef_.flatten(), dask_ml_model.coef_, rtol=1e-3, atol=1e-3)
    
    # Test that both coefs are within an epsilon-ball
    est, truth = dask_mo_model.coef_, skl_model.coef_.flatten()
    rel_error = np.norm(est - truth) / np.norm(truth)
    assert rel_error < 1e-6
```

 Relative error is a weaker condition often used in convergence proofs to ensures that both elements are close enough. I think that's what this test expresses but it'd be nice to see it more explicitly.

Reply:
Yes, the goal of this test is to ensure that the coefficients are equal with some small amount of relative error. Two differences between the `np.testing.assert_allclose` and relative error calculation that you suggest is that (1) the NumPy utility computes the error element-wise and bounds the maximum error for each element, and (2) the `atol` condition allows for an absolute tolerance when the relative tolerance comes from numeric instability (i.e. when the difference is compared to a near-zero coefficient and has high relative error but absolute error at an order of magnitude of `1e-10`).

I am happy to update the relative error to be calculated via that vector norm rather than element-wise, but wanted to call out that difference before updating the implementation.